### PR TITLE
Deploy NVHPC 22.3 and CUDA 11.6.1

### PR DESCRIPTION
**Pull request by olupton**
Cherry-picked https://github.com/spack/spack/pull/28630 and then updated CUDA and NVHPC recipes with upstream develop.

### CODE REVIEW

File: bluebrain/deployment/environments/externals.yaml [hunk 1]
```diff
@@ -63,10 +63,10 @@ spack:
     - llvm@13.0.0
     - llvm@12.0.1
     - llvm@11.1.0
-    - nvhpc@21.2 install_type=network
     - nvhpc@21.11
     - nvhpc@22.1
     - nvhpc@22.2
```
Comment (pramodk) [inline bluebrain/deployment/environments/externals.yaml]: should remove any older version?

Comment (olupton) [reply bluebrain/deployment/environments/externals.yaml]: I removed 21.2, maybe it's too far up to show in this bit of the diff.

**Comment by olupton** [comment]
I launched https://bbpgitlab.epfl.ch/hpc/coreneuron/-/pipelines/47900 using this PR's build directory.

**Comment by olupton** [comment]
> I launched https://bbpgitlab.epfl.ch/hpc/coreneuron/-/pipelines/47900 using this PR's build directory.

This pipeline was fine.

File: bluebrain/deployment/environments/externals.yaml (hunk 1 above)
Comment (pramodk) [reply bluebrain/deployment/environments/externals.yaml]: may be 22.1 can be also removed if we want to keep last stable 22.2 and new 22.3? (all older versions are in archive anyway if we want to grab them for testing/debugging)

**Review by matz-e — DISMISSED** [review]

File: bluebrain/deployment/environments/externals.yaml (hunk 1 above)
Comment (olupton) [reply bluebrain/deployment/environments/externals.yaml]: OK, I can drop 22.1 ~and CUDA 11.5.1 that went with it~ (CUDA 11.5.1 matches NVHPC 21.11 that we would still have).
Also wondering if we need CUDA 11.0.2 for any reason any more.

Comment (olupton) [reply bluebrain/deployment/environments/externals.yaml]: Dropped 22.1 in 64c794f04e.

**Review by pramodk — APPROVED** [review]

**Pull request merged**